File: backend/src/main/java/com/workforcex/backend/dto/JobRequest.java
package com.workforcex.backend.dto;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;

public record JobRequest(
        @NotBlank(message = "Title is required")
        String title,

        String skillsRequired,          // comma-separated

        Integer experienceRequired,

        // Comma-separated cities, e.g. "Mumbai,Pune,Thane"
        String location,

        Double salaryMin,
        Double salaryMax,

        @Min(value = 1, message = "At least 1 position required")
        Integer openPositions,

        String description
) {
}

File: android/employer/shared_employer/src/commonMain/kotlin/com/workforcex/shared_employer/models/JobRequest.kt
package com.workforcex.shared_employer.models

data class JobRequest(
    val title: String,
    val skillsRequired: String,
    val experienceRequired: Int,
    val location: String,
    val salaryMin: Double,
    val salaryMax: Double,
    val openPositions: Int,
    val description: String
)